bentoml/_internal/frameworks/tensorflow_v2.py

```diff
@@ -386,129 +188,166 @@ def __call__(self, inputs):
-        raw_predict_fn = getattr(self._model, self._predict_fn_name)  # type: ignore
-        self._predict_fn = functools.partial(raw_predict_fn, **self._partial_kwargs)
-
-    def _run_batch(self, *args: "TFArgType", **kwargs: "TFArgType") -> "ext.NpNDArray":
-        params = Params["TFArgType"](*args, **kwargs)
-
-        with tf.device(self._device_id):  # type: ignore
-
-            def _mapping(item: "TFArgType") -> "tf_ext.TensorLike":
-                if not LazyType["tf_ext.TensorLike"]("tf.Tensor").isinstance(item):
-                    return t.cast("tf_ext.TensorLike", tf.convert_to_tensor(item))
-                else:
-                    return item
-
-            params = params.map(_mapping)
-
-            tf.compat.v1.global_variables_initializer()  # type: ignore
-            res = self._predict_fn(*params.args, **params.kwargs)
-            return t.cast("ext.NpNDArray", res.numpy())
+        return _model.tag
 
 
-def load_runner(
-    tag: t.Union[str, Tag],
-    *,
-    predict_fn_name: str = "__call__",
-    device_id: str = "CPU:0",
-    name: t.Optional[str] = None,
-    partial_kwargs: t.Optional[t.Dict[str, t.Any]] = None,
-) -> "_TensorflowRunner":
+def get_runnable(
+    bento_model: bentoml.Model,
+):
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
     """
-    Runner represents a unit of serving logic that can be scaled horizontally to
-    maximize throughput. `bentoml.tensorflow.load_runner` implements a Runner class that
-    wrap around a Tensorflow model, which optimize it for the BentoML runtime.
-
-    Args:
-        tag (:code:`Union[str, Tag]`):
-            Tag of a saved model in BentoML local modelstore.
-        predict_fn_name (:code:`str`, default to :code:`__call__`):
-            Inference function to be used.
-        partial_kwargs (:code:`Dict[str, Any]`, `optional`, default to :code:`None`):
-            Dictionary of partial kwargs that can be shared across different model.
-        device_id (:code:`str`, `optional`, default to the first CPU):
-            Optional devices to put the given model on. Refers to `Logical Devices <https://www.tensorflow.org/api_docs/python/tf/config/list_logical_devices>`_ from TF documentation.
-
-    Returns:
-        :obj:`~bentoml._internal.runner.Runner`: Runner instances for :mod:`bentoml.tensorflow` model
-
-    Examples:
 
-    .. code-block:: python
+    partial_kwargs: t.Dict[str, t.Any] = bento_model.info.options.get(
+        "partial_kwargs", dict()
+    )
 
-        import bentoml
+    class TensorflowRunnable(Runnable):
+        SUPPORT_GPU = True
+        SUPPORT_MULTI_THREADING = True
 
-        # load a runner from a given flag
-        runner = bentoml.tensorflow.load_runner(tag)
+        def __init__(self):
+            super().__init__()
+            if len(tf.config.list_physical_devices("GPU")) > 0:
+                self.device_name = "/device:GPU:0"
```

Comment:
```suggestion
                # In Multi-GPU scenarios, the visible cuda devices will be set for each Runner worker
                # by the runner's Scheduling Strategy. So that the Runnable implementation only needs 
                # to find the first GPU device visible to current process.
                self.device_name = "/device:GPU:0"
```

Reply:
done